.github/workflows/treadmill-ci-test.yml

```diff
@@ -1,28 +1,23 @@
-name: treadmill-ci-test
+name: treadmill-ci-dev-branch
 
 env:
   TERM: xterm # Makes tput work in actions output
 
-# Controls when the action will run. Triggers the workflow on pull request and
-# merge group checks:
-#
-# KEEP IN SYNC WITH `environment:` ATTRIBUTE BELOW:
+# Controls when the action will run. Triggers the workflow on commits to the dev branch
 on:
+  # Regular CI on the long-lived dev branch
   push:
     branches:
-      - main
-  # Pull requests from forks will not have access to the required GitHub API
-  # secrets below, even if they are using an appropriate deployment environment
-  # and the workflow runs have been approved according to this environment's
-  # rules. We don't know whether this is a bug on GitHub's end or deliberate.
-  # Either way, for now we disable this workflow to run on PRs until we have
-  # an API proxy that securely performs these GitHub API calls (adding runners
-  # and starting Treadmill jobs with those runner registration tokens), which
-  # allows this workflow to run without access to repository secrets.
-  #pull_request:
-  merge_group: # Run CI for the GitHub merge queue
-  # Manually dispatch for a specific branch (will require approval
-  # through the treadmill-ci-merged environment:
+      - dev/multiple_board_thci_clean
```

Comment:
I'm confused about what's going on here. Is this a special, long-lived branch that should always be used for.... something? If so, what does "thci_clean" mean? It should be documented somewhere where we won't forget that we should use this branch for whatever the right purpose is.